partner_firstname/models/partner.py

```diff
@@ -90,21 +131,23 @@ def _get_inverse_name(self, name, is_company=False):
             parts = [name or False, False]
         # Guess name splitting
         else:
-            parts = name.split(" ", 1)
+            parts = name.strip().split(" ", 1)
             while len(parts) < 2:
                 parts.append(False)
-        return parts
+        return {"lastname": parts[0], "firstname": parts[1]}
 
     @api.one
     def _inverse_name(self):
+        """Try to revert the effect of :meth:`._compute_name`."""
         parts = self._get_inverse_name(self.name, self.is_company)
-        self.lastname, self.firstname = parts
+        self.lastname, self.firstname = parts["lastname"], parts["firstname"]
 
     @api.one
     @api.constrains("firstname", "lastname")
     def _check_name(self):
         """Ensure at least one name is set."""
-        if not (self.firstname or self.lastname):
+        if ((self.type == 'contact' or self.is_company) and
+                not (self.firstname or self.lastname)):
```

Comment:
Yup it isn't required anymore as you can have a partner as a shipping address or an invoicing address without name.